demo-java/app/src/gms/java/com/example/placesdemo/AutocompleteAddressActivity.java

```diff
@@ -0,0 +1,259 @@
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MapStyleOptions;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.libraries.places.api.Places;
+import com.google.android.libraries.places.api.model.AddressComponent;
+import com.google.android.libraries.places.api.model.AddressComponents;
+import com.google.android.libraries.places.api.model.Place;
+import com.google.android.libraries.places.api.net.PlacesClient;
+import com.google.android.libraries.places.widget.Autocomplete;
+import com.google.android.libraries.places.widget.AutocompleteActivity;
+import com.google.android.libraries.places.widget.model.AutocompleteActivityMode;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Activity for using Place Autocomplete to assist filling out an address form.
+ */
+@SuppressWarnings("FieldCanBeLocal")
+public class AutocompleteAddressActivity extends AppCompatActivity implements OnMapReadyCallback {
+
+    private static final String TAG = "ADDRESSAUTOCOMPLETE";
+    private static final String MAP_FRAGMENT_TAG = "map";
+    private static final int AUTOCOMPLETE_REQUEST_CODE = 23487;
+    private EditText address1Field;
+    private EditText address2Field;
+    private EditText cityField;
+    private EditText stateField;
+    private EditText postalField;
+    private EditText countryField;
+    private LatLng coordinates;
+    private SupportMapFragment mapFragment;
+    private GoogleMap map;
+    private Marker marker;
+    private PlacesClient placesClient;
+    private View mapPanel;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Use whatever theme was set from the MainActivity - some of these colors (e.g primary color)
+        // will get picked up by the AutocompleteActivity.
+        int theme = getIntent().getIntExtra(MainActivity.THEME_RES_ID_EXTRA, 0);
+        if (theme != 0) {
+            setTheme(theme);
+        }
+
+        setContentView(R.layout.autocomplete_address_activity);
+
+        // Retrieve a PlacesClient (previously initialized - see MainActivity)
+        placesClient = Places.createClient(this);
+
+        address1Field = (EditText) findViewById(R.id.autocomplete_address1);
+        address2Field = (EditText) findViewById(R.id.autocomplete_address2);
+        cityField = (EditText) findViewById(R.id.autocomplete_city);
+        stateField = (EditText) findViewById(R.id.autocomplete_state);
+        postalField = (EditText) findViewById(R.id.autocomplete_postal);
+        countryField = (EditText) findViewById(R.id.autocomplete_country);
+
+        // Attach an Autocomplete intent to the Address 1 EditText field
+        address1Field.setOnTouchListener((v, event) -> {
```

Comment:
This may be an awkward pattern (opening an AC intent when the user touches Address1 field). Is there a better way of using the Intent method for adding AC widget? @arriolac

Reply:
Do you have a suggestion for how to prevent double opening of autocomplete intents from double-tapping? Is there a way to make sure that max one intent exists at any given time?

We don't want AC to open on other fields - it would be too intrusive for a user who is trying to manually enter their address.

Reply:
> Do you have a suggestion for how to prevent double opening of autocomplete intents from double-tapping? Is there a way to make sure that max one intent exists at any given time?

You can try disabling the view (using `setEnabled(false)`) which I think should stop touch events from being received, you can then reenable the view again in `onActivityResult`.